docs/Client.asciidoc

```diff
@@ -0,0 +1,320 @@
+be handy if for example the REST endpoint you are using supports content
+negotiation.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com -a 'Accept: application/json' \
+    jobs/overview
+----
+
+=== HTTP Body
+
+To change the HTTP request body there are multiple options available. The
+simplest being `--data` (or `-d` for short), which allows you to use a plain
+string as request body. This can be useful for example to change the group id of
+a job.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com -X PUT jobs/1 \
+    --data '{"group_id":2}'
+----
+
+With the `--data-file` option (or `-D` for short) you can also use a file
+instead.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com -X PUT jobs/1 \
+    --data-file ./test.json
+----
+
+Or just pipe the data to `openqa-cli`.
+
+[source,sh]
+----
+echo '{"group_id":2}' | openqa-cli api --host http://openqa.example.com -X PUT \
+    jobs/1
+----
+
+=== Forms
+
+Most data you pass to the openQA API will be key/value form parameters. Either
+in the query string, or encoded as `application/x-www-form-urlencoded` HTTP
+request body. But you don't have to worry about this too much, because
+`openqa-cli api` knows when to use which format automatically, you just provide
+the key/value pairs.
+
+Form parameters are most commonly passed as additional arguments after the path.
+For example to post a comment to a job.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com -X POST jobs/2/comments text=abc
+----
+
+This value can also be quoted to include whitespace characters.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com -X POST jobs/2/comments \
+    text="Hello openQA!"
+----
+
+And you can use interpolation to include files.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com -X POST jobs/2/comments \
+    text="$(cat ./comment.markdown)"
+----
+
+Alternatively you can also use the `--form` option (or `-f` for short) to
+provide all form parameters in JSON format. Here you would reuse the HTTP body
+options, such as `--data` and `--data-file`, to pass the JSON document to be
+turned into form parameters.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com --form --data '{"text":"abc"}' \
+    -X POST jobs/2/comments
+----
+
+=== JSON
+
+The primary data exchange format in the openQA API is JSON. And you will even
+see error messages in JSON format most of the time.
+
+[source,json]
+----
+{"error":"no api key","error_status":403}
+----
+
+By default the returned JSON is often compressed, for better performance, and
+can be hard to read if the response gets larger. But if you add the `--pretty`
+option (or `-p` for short), `openqa-cli` can reformat it for you into much more
+user friendy format, with newlines and indentation.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com --pretty jobs/overview
+----
```

Comment:
An example would be nice, like above, to demonstrate the nicer looking JSON output.